### PR TITLE
fix for Task.spec.inputs.params when using NGP

### PR DESCRIPTION
**Pull request by jstrachan**

### CODE REVIEW

File: pkg/jx/cmd/clients/factory.go [hunk 1]
```diff
@@ -382,13 +383,45 @@ func (f *factory) ResetSecretsLocation() {
...
 }
 
+func (f *factory) getVaultName(namespace string) (string, error) {
+	name := ""
+	context, err := cluster.Context(f.kubeConfig)
+	if err != nil {
+		return name, err
+	}
+	if context == nil {
+		kubeClient, _, err := f.CreateKubeClient()
```
Comment (ccojocar) [inline pkg/jx/cmd/clients/factory.go]: Is it the client created successfully if the context is nil?

File: pkg/jx/cmd/clients/factory.go [hunk 2]
```diff
@@ -382,13 +383,45 @@ func (f *factory) ResetSecretsLocation() {
...
+		if err != nil {
+			return name, err
+		}
+		cm, err := kube.GetConfigMap(kubeClient, namespace, kube.ConfigMapNameJXInstallConfig)
```
Comment (ccojocar) [inline pkg/jx/cmd/clients/factory.go]: There is a helper function for this `kube.ReadInstallValues`.

File: pkg/jx/cmd/clients/factory.go [hunk 3]
```diff
@@ -382,13 +383,45 @@ func (f *factory) ResetSecretsLocation() {
...
+		if err != nil {
+			return name, errors.Wrapf(err, "no ConfigMap %s in namespace %s", kube.ConfigMapNameJXInstallConfig, namespace)
+		}
+		if cm.Data == nil {
+			return name, fmt.Errorf("no data for ConfigMap %s in namespace %s", kube.ConfigMapNameJXInstallConfig, namespace)
+		}
+		name = cm.Data["clusterName"]
```
Comment (ccojocar) [inline pkg/jx/cmd/clients/factory.go]: This const is defined in `kube.ClusterName`.

Comment (jstrachan) [reply pkg/jx/cmd/clients/factory.go]: lemme use that

File: pkg/jx/cmd/clients/factory.go (hunk 1 above)
Comment (jstrachan) [reply pkg/jx/cmd/clients/factory.go]: if there's no config then we can still create a k8s client yeah: https://github.com/jenkins-x/jx/blob/976b6e8d676ed1c6bfda05db85b3727876769c8a/pkg/jx/cmd/clients/factory.go#L617

**Pull request merged**